app/src/main/res/layout/activity_main.xml

```diff
@@ -267,6 +270,39 @@
 
             </LinearLayout>
 
+            <View
+                android:layout_width="1dp"
+                android:layout_height="match_parent"
+                android:background="@drawable/menu_line_gradient_bg"/>
+
+            <LinearLayout
+                android:id="@+id/line_detection_container"
+                android:layout_width="0dp"
+                android:layout_height="wrap_content"
+                android:layout_marginBottom="17dp"
+                android:layout_weight="1"
+                android:gravity="center_horizontal"
+                android:orientation="vertical">
+
+                <ImageView
+                    android:id="@+id/line_detection"
+                    android:layout_width="68dp"
```

Comment:
Here and in other places - to `dimens.xml`.

Reply:
I don't see the advantage of using dimens everywhere. IMHO it creates more mess rather than profit. You change one dimen and accidentally got troubles in another place.
The only good scenario I saw was styling for themes, it really good place to make all desing unified and pixel perfect.

Reply:
From another point, it is a new project for me with own style guides which you established before, and I'll change if required, just would be nice to discuss

Reply:
I'd prefer to keep not super common dimens in layouts. We should have a good & clear naming for all these dimen components too and iterate often over the design to get real profit from common components. 

But seems the design is not a key component of Teaser and not often changed too